httpbakery/keyring_test.go

```diff
@@ -104,7 +104,7 @@ func (s *KeyringSuite) TestPublicKeyWrongURL(c *gc.C) {
 	client := httpbakery.NewHTTPClient()
 	_, err := httpbakery.PublicKeyForLocation(client, "http://localhost:0")
 	c.Assert(err, gc.ErrorMatches,
-		`cannot get public key from "http://localhost:0/publickey": Get http://localhost:0/publickey: dial tcp 127.0.0.1:0: .*connection refused`)
+		`cannot get public key from "http://localhost:0/publickey": Get http://localhost:0/publickey: dial tcp (127.0.0.1|\[::1\]):0: .*connection refused`)
```

Comment:
or just `dial tcp .*: ` to make us resilient against future similar changes?

Reply:
do you expect there will be a IPv7 anytime soon?
I wanted to keep the explict 127 or ::1 to be clear that we are talking about localhost.

Reply:
depends whether you want to care about the IP address. If all you care'd about was the "cannot get public key" then why have any of the trailing information?
`cannot get public key from "http://.*".*`

I don't really know what the goal of the test is. I'm happy to be prompted in a different direction.

Reply:
We don't care about the details of the message, but it's nice to know that it happens because the connection was refused rather than some other random reason.